src/ppx/application.cpp

```diff
@@ -788,13 +788,18 @@ void Application::ShutdownMetrics()
 
 void Application::SaveMetricsReportToDisk()
 {
-    if (!mStandardOpts.pEnableMetrics->GetValue()) {
+    if (mStandardOpts.pEnableMetrics == nullptr || !mStandardOpts.pEnableMetrics->GetValue()) {
         return;
     }
 
+    std::string metricsFilename;
+    if (mStandardOpts.pMetricsFilename != nullptr) {
```

Comment:
Same question of whether this can be nullptr. Also this has a default value of report_@, the one error I ran into is in Report::WriteToDisk when there parent_path is empty with the default value, and it throws an error. I think that's something we need to fix as well.

Reply:
I'd like to restore the functionality that was removed during the Knob transition. Can the `WriteToDisk` issue be addressed in a follow-up PR?

The assertions should catch if `nullptr` finds its way back into this function.

Reply:
Sure!